base/test.jl

```diff
@@ -407,11 +358,27 @@ function finish(ts::DefaultTestSet)
     println()
     # Recursively print a summary at every level
     print_counts(ts, 0, align, pass_width, fail_width, error_width, total_width)
-    # Finally throw an error as we are the outermost test set
-    if total != total_pass
-        throw(TestSetException(total_pass,total_fail,total_error))
-    end
+end
 
+# store the results in the enclosing testset and the top level result stream
```

Comment:
Yeah thats what I meant - it just made the diff in this commit larger than it needed to be, looks like a bigger change